src/buffer.h

```diff
@@ -108,6 +108,7 @@ class SubBuffer : public Buffer {
       : Buffer(size),
         _data(src, src->data(offset))
     {
+      ASSERT(offset.v, <=, src->size().v);
```

Comment:
Shouldn't we have `offset < src->size()` if `offset + size < src->size()` ?

Reply:
Don't forget about overflow. A simple counterexample is `offset == offset_type(-1)`.